### PR TITLE
Contributing guidelines

### PR DESCRIPTION
**Pull request by carlessistare**
Some guidelines for contributors and code reviewers to be followed

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+* Make sure you have a [GitHub account](https://github.com/signup/free)
+* Submit a ticket for your issue, assuming one does not already exist.
+  * Clearly describe the issue including steps to reproduce when it is a bug.
+  * Make sure you fill in the earliest version that you know has the issue.
+* Fork the repository on GitHub
+
+## Making Changes
+
+* Create a topic branch from where you want to base your work.
```
Comment (estliberitas) [inline CONTRIBUTING.md]: Well, non-contributers won't be able to create topic branch. They will have to fork.

Comment (estliberitas) [inline CONTRIBUTING.md]: It's better to make everybody use forks and create branches there instead of trashing main repo

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  * This is usually the master branch.
+  * Please avoid working directly on the `master` branch.
```
Comment (estliberitas) [inline CONTRIBUTING.md]: I think this would a crime at all

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+* The code style of current code base should be preserved
+* Make sure you have added the necessary tests for your changes, specially if
+you added a new feature.
+* Run _all_ the tests to assure nothing else was accidentally broken.
+
+## Submitting Changes
+
+* Push your changes to a topic branch in your fork of the repository.
+* Submit a pull request to the repository.
+* Make sure that the PR has a clean log message.
```
Comment (estliberitas) [inline CONTRIBUTING.md]: Maybe recommend to squash commits? That would help with clean commit history.

Comment (hyperlink) [inline CONTRIBUTING.md]: Didn't github just add squashed merges?

**Comment by estliberitas** [comment]
Until @haio enables rebase-like merge, I recommend we use rebase instead of merge. https://github.com/blog/2141-squash-your-commits

**Comment by carlessistare** [comment]
@estliberitas  I agree, I added a comment with this.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+* Make sure you have added the necessary tests for your changes, specially if
+you added a new feature.
+* Run _all_ the tests to assure nothing else was accidentally broken.
+
+## Submitting Changes
+
+* Push your changes to a topic branch in your fork of the repository.
+* Submit a pull request to the repository.
+* Make sure that the PR has a clean log message and don't hesitate to squash 
+and rebase your commits in order to preserve a clean hitory log.
```
Comment (ultrafez) [inline CONTRIBUTING.md]: There's a typo in the word "history" here

**Comment by estliberitas** [comment]
@haio Will you add squash style merging soon? I'd like to use it )

**Comment by carlessistare** [comment]
The problem of this, is that we cannot easily revert PR am I wrong?
I usually use squash style to merge differet feature PRed in a feature
branch, but not in master, in order to be able unmerge in case of problems.
What do you think?

**Comment by estliberitas** [comment]
@carlessistare Well, I did not see if github does it for this new feature. If it does not have "Revert" button, then it will be a kinda issue but not that big one.

**Comment by hyperlink** [comment]
Squash merge is available.

image

**Pull request merged**